third_party/move/testing-infra/transactional-test-runner/Cargo.toml

```diff
@@ -11,6 +11,7 @@ edition = "2021"
 
 [dependencies]
 anyhow = { workspace = true }
+aptos-framework = { workspace = true, optional = true }
```

Comment:
Before we never included aptos dependencies for third party, maybe this can be passed as a param? Though for framework / testing current structure is really inconvenient, we only get access to legacy stdlib....

Reply:
Yes, these boundaries are rather artificial and hard to work with actually (kills the purpose), and we are not really doing anything with third party ... cc @vgao1996 @wrwg

Reply:
Yeah this has been a long-standing issue. The original reasoning is to keep third-party/move platform-agnostic, but I also doubt if this still makes sense any more. Personally I'm supportive of us prioritizing unblocking our own work here, that is, go ahead and depend on aptos crates if needed. 

I'm facing a similar issue with the package system -- the package resolver need to be aware of on-chain (Aptos) dependencies, otherwise it couldn't fetch them. @wrwg  and I discussed this and we decided it's probably not worth introducing additional traits/callbacks just to avoid this explicit dependency, at least for now.